src/assets/src/components/common.tsx

```diff
@@ -274,38 +271,45 @@ interface SingleInputFormShowRemainingProps extends StatelessSingleInputFormProp
 }
 
 export const SingleInputFormShowRemaining: React.FC<SingleInputFormShowRemainingProps> = (props) => {
-    const inputRef = createRef<HTMLTextAreaElement>();
-    useEffect(() => {
-        if (!props.autofocus) return;
-        inputRef.current!.focus();
-    }, []);
-    const submit = (e: React.FormEvent<HTMLFormElement>) => {
-        e.preventDefault();
-        props.onSubmit(props.value);
-        props.onChangeValue("");
+    const [remaining, setRemaining] = useState(props.maxLength - props.value.length as number)
+    const [isInvalid, setIsInvalid] = useState(false as boolean)
```

Comment:
I think `isValid` can be derived statelessly from `remaining`? And although I didn't notice it before, it looks like maybe there's no need for state `remaining` either, as that can be derived from `props.maxLength` and `props.value.length`. So I think these two can be replaced with:
```
const remaining = props.maxLength - props.value.length;
```
and
```
const isInvalid = remaining < 0
```
All of the state updates could then be removed.
Let me know if I'm overlooking something.

Reply:
Hm, good point, I'll take a look and simplify if possible. Thanks!

Reply:
Done. Makes more sense to me now that I look back and see the higher-level `ShowRemainingField`.